test/lexer/numbers.ts

```diff
@@ -151,15 +151,21 @@ describe('Lexer - Numberic literals', () => {
     [Context.None, Token.NumericLiteral, '0000000000234567454548', 234567454548],
 
     // BigInt
-    [Context.None, Token.BigIntLiteral, '1n', 1],
-    [Context.None, Token.BigIntLiteral, '0o45n', 37],
-    [Context.None, Token.BigIntLiteral, '0b10n', 2],
-    [Context.None, Token.BigIntLiteral, '0x9an', 154],
-    [Context.None, Token.BigIntLiteral, '9007199254740991n', 9007199254740991],
-    [Context.None, Token.BigIntLiteral, '100000000000000000n', 100000000000000000],
-    [Context.OptionsRaw, Token.BigIntLiteral, '123456789000000000000000n', 123456789000000000000000],
```

Comment:
Note `123456789000000000000000` is bigger than `Number.MAX_SAFE_INTEGER`. This existing test was passed by double bugs:
1. the stored tokenValue is wrong
2. the logic to check tokenValue againt js number `123456789000000000000000` doesn't work as expectation, because of losing precision.